File: app/src/main/java/com/example/andreip/myapplication/app/Recipe.java
package com.example.andreip.myapplication.app;

import com.google.gson.annotations.SerializedName;

import SQLData.RecipeDb;

public class Recipe {

    @SerializedName("title")
    private String title;
    @SerializedName("href")
    private String href;
    @SerializedName("ingredients")
    private String ingredients;
    @SerializedName("thumbnail")
    private String thumbnail;

    public Recipe(RecipeDb recipeDb) {
        this.title = recipeDb.getTitle();
        this.href = recipeDb.getHref();
        this.ingredients = recipeDb.getIngredients();
        this.thumbnail = recipeDb.getThumbnail();
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setHref(String href) {
        this.href = href;
    }

    public String getHref() {
        return this.href;
    }

    public void setIngredients(String ingredients) {
        this.ingredients = ingredients;
    }

    public String getIngredients() {
        return this.ingredients;
    }

    public String getThumbnail() {
        return this.thumbnail;
    }

    public void setThumbnail(String thumbnail) {
        this.thumbnail = thumbnail;
    }


}

File: app/src/main/java/com/example/andreip/myapplication/app/Recipe.kt
package com.example.andreip.myapplication.app

import SQLData.RecipeDb
import com.google.gson.annotations.SerializedName

class Recipe() {

    @SerializedName("title")
    private var title: String? = null
    @SerializedName("href")
    private var href: String? = null
    @SerializedName("ingredients")
    private var ingredients: String? = null
    @SerializedName("thumbnail")
    private var thumbnail: String? = null

    constructor (recipeDb: RecipeDb) : this() {
        this.title = recipeDb.title;
        this.href = recipeDb.href;
        this.ingredients = recipeDb.ingredients;
        this.thumbnail = recipeDb.thumbnail;

    }

    fun getTitle(): String? {
        return title
    }

    fun setTitle(title: String) {
        this.title = title
    }

    fun setHref(href: String) {
        this.href = href
    }

    fun getHref(): String? {
        return this.href
    }

    fun setIngredients(ingredients: String) {
        this.ingredients = ingredients
    }

    fun getIngredients(): String? {
        return this.ingredients
    }

    fun getThumbnail(): String? {
        return this.thumbnail
    }

    fun setThumbnail(thumbnail: String) {
        this.thumbnail = thumbnail
    }

}
